src/webloader.js

```diff
@@ -53,32 +53,25 @@ function injectStylesheet(endpoint, env, version, uid) {
 	document.head.appendChild(stylesheet);
 }
 
-function handlePushstate(client) {
+function handlePushState(client) {
 	const pushStateOrig = history.pushState;
+	const replaceStateOrig = history.replaceState;
 
-	history.pushState = function() {
-		const args = Array.prototype.slice.call(arguments);
-		pushStateOrig.apply(history, args);
-	
-		let event;
-		const eventType = 'pushstate_evolv';
-	
-		if (Event.prototype.constructor) {
-			event = new CustomEvent(eventType, {});
-		} else { // For IE Compatibility
-			event = document.createEvent('Event');
-			event.initEvent(eventType);
-		}
-	
-		window.dispatchEvent(event);
-	};
-
-	const updateContext = function () {
+	function updateContext() {
 		client.context.set('web.url', window.location.href);
-	};
+	}
+
+	function handler(orig) {
+		const args = Array.prototype.slice.call(arguments, 1);
+		orig.apply(history, args);
+
+		updateContext();
+	}
+
+	history.pushState = handler.bind(this, pushStateOrig);
+	history.replaceState = handler.bind(this, replaceStateOrig);
```

Comment:
> This commit also removes the global pushstate_evolv event, which polutes the
global state and should not be used. If updates on url need to be listened
to, subscribing to context.changed is the correct solution.

I disagree with the assertion that the former solution pollutes the global namespace in a way different than this proposed solution. In fact, it was implemented in the way that it was to minimize modification of the native behavior. The event dispatch creates an execution boundary so that any bugs in the asset manager that crop up do not corrupt the behavior of `history.pushState()`. Other than the event dispatch, the two solutions override the native behavior equally.

Was there an express need to make this modification?

Reply:
So, there are a couple of things here. The previous design leaked internal implementation, and exposed a public interface for something we shouldn't be doing in the first place, in which the only valid consumer was a few lines away. There is a document public interface for triggering functionality when our context updates, and everyone who cares should listen to that.

To avoid any potential interrupt from exceptions in the evolv code, we can add a setTimeout(..., 0) to updating the context.